### PR TITLE
TEST: Add llama logits tests

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

Adds the llama logits tests that have been always skipped 😢 

Tested on our A10 & T4 Runners !

### CODE REVIEW

File: tests/models/llama/test_modeling_llama.py [hunk 1]
```diff
@@ -605,76 +603,79 @@ def setUpClass(cls):
...
-        model = LlamaForCausalLM.from_pretrained("meta-llama/Llama-2-13b-hf", device_map="auto")
-        out = model(torch.tensor(input_ids))
-        # Expected mean on dim = -1
-        EXPECTED_MEAN = torch.tensor([[-2.0622, -1.2794, -1.1638, -0.9788, -1.4603, -1.0238, -1.7893, -1.4411]])
-        torch.testing.assert_close(out.mean(-1), EXPECTED_MEAN, atol=1e-2, rtol=1e-2)
-        # slicing logits[0, 0, 0:30]
-        EXPECTED_SLICE = torch.tensor([-8.1406, -8.0547, 2.7461, -1.2344, -0.1448, -1.8262, -1.0020, -1.8154, -1.6895, -1.8516, -2.3574, -0.9277, 3.7598, 6.5742, -1.2998, -0.1177, -8.1406, -2.9688, -2.9199, -3.1699, -3.5254, -2.3555, -2.7988, -3.4141, -2.8262, -4.5195, -3.3379, -3.3164, -2.7832, -3.0273])  # fmt: skip
-        torch.testing.assert_close(out[0, 0, :30], EXPECTED_SLICE, atol=1e-5, rtol=1e-5)
+        model = LlamaForCausalLM.from_pretrained(
+            "meta-llama/Llama-2-7b-hf", device_map="auto", torch_dtype=torch.bfloat16, attn_implementation="eager"
```
Comment (younesbelkada) [inline tests/models/llama/test_modeling_llama.py:612]: we need to use eager otherwise bf16 + SDPA fails

File: tests/models/llama/test_modeling_llama.py [hunk 2]
```diff
@@ -605,76 +603,79 @@ def setUpClass(cls):
...
+            8: torch.tensor([[-6.6544, -4.1259, -4.9840, -3.2456,  0.8261, -3.0124,  1.2971, -3.3641]]) # fmt: skip
+        }
+
+        self.assertTrue(torch.allclose(EXPECTED_MEAN[self.cuda_compute_capability_major_version].to(torch_device), out.logits.mean(-1), atol=1e-2, rtol=1e-2))
+
+        # slicing logits[0, 0, 0:15]
+        EXPECTED_SLICE = {
+            7: torch.tensor([-12.8125, -7.3359, -0.4846, -8.0234, -7.2383, -7.9922, -6.4805, -7.7344, -7.8125, -7.0078, -6.1797, -7.1094, -1.8633, 1.9736, -8.6016]), # fmt: skip
+            8: torch.tensor([-12.8281,  -7.4609,  -0.4668,  -8.0703,  -7.2539,  -8.0078,  -6.4961, -7.7734,  -7.8516,  -7.0352,  -6.2188,  -7.1367,  -1.8564,   1.9922, -8.6328]) # fmt: skip
+        }
```
Comment (ArthurZucker) [inline tests/models/llama/test_modeling_llama.py:668]: fmt skip not needed in between fmt on and off

File: tests/models/llama/test_modeling_llama.py [hunk 3]
```diff
@@ -736,89 +737,6 @@ def test_compile_static_cache(self):
...
-    refl
-}
-end
-""",
-    ]
-
-    @require_torch_accelerator
-    @slow
-    @unittest.skip("Model is too large")
-    def test_model_7b_logits(self):
```
Comment (ArthurZucker) [inline tests/models/llama/test_modeling_llama.py:782]: As long as the tokenizer is tested with this, LGMT to remove

**Review by ArthurZucker — COMMENTED** [review]
Can you add one commit with `[run-slow]` to check it's all good?

**Review by ArthurZucker — APPROVED** [review]

File: tests/models/llama/test_modeling_llama.py (hunk 2 above)
Comment (younesbelkada) [reply tests/models/llama/test_modeling_llama.py]: nice catch!

File: tests/models/llama/test_modeling_llama.py [hunk 4]
```diff
@@ -605,76 +603,79 @@ def setUpClass(cls):
...
 
-        EXPECTED_MEAN = torch.tensor(
-            [[-4.2327, -3.3360, -4.6665, -4.7631, -1.8180, -3.4170, -1.4211, -3.1810]], dtype=torch.float32
+        # fmt: off
+        EXPECTED_MEAN = {
+            7: torch.tensor([[-6.5061, -4.1147, -4.9669, -3.2038, 0.8069, -2.9694, 1.2864, -3.3786]]),
+            8: torch.tensor([[-6.5208, -4.1218, -4.9377, -3.2536,  0.8127, -2.9811,  1.2918, -3.3848]])
+        }
+
+        self.assertTrue(torch.allclose(EXPECTED_MEAN.to(torch_device), out.logits.mean(-1), atol=1e-2, rtol=1e-2))
```
Comment (younesbelkada) [inline tests/models/llama/test_modeling_llama.py:625]:
```suggestion
        self.assertTrue(torch.allclose(EXPECTED_MEAN[self.cuda_compute_capability_major_version].to(torch_device), out.logits.mean(-1), atol=1e-2, rtol=1e-2))
```
